### PR TITLE
Bump gVisor to 20230911.0

### LINKED ISSUE #683: Bump gVisor to 20230911.0
A regular upgrade to keep up with upstream.

### PR DESCRIPTION
**Pull request by CuriousGeorgiy**
## Summary

Closes #683 

## Implementation Notes :hammer_and_pick:

* N/A.

## External Dependencies :four_leaf_clover:

* Bumped gVisor to 20230911.0.

## Breaking API Changes :warning:

* N/A.

### CODE REVIEW

**Comment by CuriousGeorgiy** [comment]
The gVisor containerd quick start guide mentions[^1] the following lines in the [configuration file](https://github.com/vhive-serverless/vHive/blob/main/configs/gvisor-containerd/config.toml):
```toml
[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc]
  runtime_type = "io.containerd.runc.v2"
```

We don't have them in our configuration file — should we?

[^1]: https://gvisor.dev/docs/user_guide/containerd/quick_start/

**Comment by anshalshukla** [comment]
Did you try running it using the quickstart guide?

**Comment by CuriousGeorgiy** [comment]
@anshalshukla Yes, I did, everything works.

**Comment by ustiugov** [comment]
@CuriousGeorgiy 1 reviewer at a time, please. @leokondrashov can you do this?

@CuriousGeorgiy I don't know, can you please check if our config is up to date? Please note that vHive has 2 containerd daemons co-running: one for containers and one for MicroVMs (gvisor VMs in this case) 
> [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc]
  runtime_type = "io.containerd.runc.v2"

**Comment by CuriousGeorgiy** [comment]
@ustiugov > I don't know, can you please check if our config is up to date?

I guess @alexandrinapanfil was following [this](https://github.com/google/gvisor-containerd-shim/blob/master/docs/runtime-handler-shim-v2-quickstart.md#configure-containerd) currently outdated guide when bringing gVisor to vHive:
> ```console
>cat <<EOF | sudo tee /etc/containerd/config.toml
>disabled_plugins = ["restart"]
>[plugins.linux]
>  shim_debug = true
>[plugins.cri.containerd.runtimes.runsc]
>  runtime_type = "io.containerd.runsc.v1"
>EOF
>```

I will try adding the lines from the newer guide.

**Comment by leokondrashov** [comment]
I tried the PR on two xl170 nodes (master + 1 worker), used multinode setup from quickstart guide, adding the gvisor as runtime to necessary scripts:

```
# master
git clone --depth=1 https://github.com/vhive-serverless/vhive.git
cd vhive
git remote add CuriousGeorgiy https://github.com/CuriousGeorgiy/vhive.git
git fetch CuriousGeorgiy
git checkout CuriousGeorgiy/gh-683-upgrade-gvisor

mkdir -p /tmp/vhive-logs

./scripts/cloudlab/setup_node.sh gvisor > >(tee -a /tmp/vhive-logs/setup_node.stdout) 2> >(tee -a /tmp/vhive-logs/setup_node.stderr >&2)

sudo screen -dmS containerd bash -c "containerd > >(tee -a /tmp/vhive-logs/containerd.stdout) 2> >(tee -a /tmp/vhive-logs/containerd.stderr >&2)"
./scripts/cluster/create_multinode_cluster.sh gvisor > >(tee -a /tmp/vhive-logs/create_multinode_cluster.stdout) 2> >(tee -a /tmp/vhive-logs/create_multinode_cluster.stderr >&2)

# worker
git clone --depth=1 https://github.com/vhive-serverless/vhive.git
cd vhive
git remote add CuriousGeorgiy https://github.com/CuriousGeorgiy/vhive.git
git fetch CuriousGeorgiy
git checkout CuriousGeorgiy/gh-683-upgrade-gvisor

mkdir -p /tmp/vhive-logs

./scripts/cloudlab/setup_node.sh gvisor > >(tee -a /tmp/vhive-logs/setup_node.stdout) 2> >(tee -a /tmp/vhive-logs/setup_node.stderr >&2)

./scripts/cluster/setup_worker_kubelet.sh gvisor > >(tee -a /tmp/vhive-logs/setup_worker_kubelet.stdout) 2> >(tee -a /tmp/vhive-logs/setup_worker_kubelet.stderr >&2)
sudo screen -dmS containerd bash -c "containerd > >(tee -a /tmp/vhive-logs/containerd.stdout) 2> >(tee -a /tmp/vhive-logs/containerd.stderr >&2)"
sudo PATH=$PATH screen -dmS gvisor bash -c "/usr/local/bin/gvisor-containerd --address /run/gvisor-containerd/gvisor-containerd.sock --config /etc/gvisor-containerd/config.toml > >(tee -a /tmp/vhive-logs/gvisor.stdout) 2> >(tee -a /tmp/vhive-logs/gvisor.stderr >&2)"
source /etc/profile && go build
sudo screen -dmS vhive bash -c "./vhive -sandbox gvisor > >(tee -a /tmp/vhive-logs/vhive.stdout) 2> >(tee -a /tmp/vhive-logs/vhive.stderr >&2)"

sudo kubeadm join #...
```

In a result, deployment fails `Error: Internal error occurred: failed calling webhook "webhook.serving.knative.dev": failed to call webhook: Post "https://webhook.knative-serving.svc:443/defaulting?timeout=10s": dial tcp 10.99.117.119:443: connect: no route to host`

Istio pods are 0/1 ready with error in the log `StreamAggregatedResources gRPC config stream closed since 580s ago: 14, connection error: desc = "transport: Error while dialing dial tcp: lookup istiod.istio-system.svc on 10.96.0.10:53: read udp 10.22.0.6:46321->10.96.0.10:53: read: connection refused"`. Seems that DNS inside cluster doesn't work. Example from here is also fails with timeout https://kubernetes.io/docs/tasks/administer-cluster/dns-debugging-resolution/.

I'm not sure that it is the issue with update or this thing happened before with gVisor deployment in general.

@CuriousGeorgiy, what initialization process did you use for testing?

**Comment by CuriousGeorgiy** [comment]
@leokondrashov I used a single node setup, i.e., `scripts/setup_node.sh gvisor` + `scripts/cloudlab/start_onenode_vhive_cluster.sh gvisor`.

**Comment by ustiugov** [comment]
please double check with 2 nodes as well when Georgiyu finishes upgrading the config

**Comment by ustiugov** [comment]
@CuriousGeorgiy you should keep yourself as an assignee of the PR but request and re-request a review from Leonid.

**Comment by CuriousGeorgiy** [comment]
I am following the standard clutser setup procedure described in quickstart guide, and I keep get the following errors with gVisor setup (both main branch and my branch).

`kubectl logs istio-ingressgateway-f5b59cc7c-n5t47 -n istio-system`:
```console
2023-03-02T10:54:33.711303Z	warning	envoy config	StreamAggregatedResources gRPC config stream closed since 274s ago: 14, connection error: desc = "transport: Error while dialing dial tcp: lookup istiod.istio-system.svc on 10.96.0.10:53: read udp 10.22.0.5:37612->10.96.0.10:53: read: connection refused"
2023-03-02T10:55:07.274865Z	warning	envoy config	StreamAggregatedResources gRPC config stream closed since 308s ago: 14, connection error: desc = "transport: Error while dialing dial tcp: i/o timeout"
2023-03-02T10:55:41.930535Z	warn	ca	ca request failed, starting attempt 1 in 91.816745ms
2023-03-02T10:55:41.930899Z	warning	envoy config	StreamAggregatedResources gRPC config stream closed since 342s ago: 14, connection error: desc = "transport: Error while dialing dial tcp: lookup istiod.istio-system.svc on 10.96.0.10:53: read udp 10.22.0.5:47418->10.96.0.10:53: read: connection refused"
2023-03-02T10:55:42.023154Z	warn	ca	ca request failed, starting attempt 2 in 199.725679ms
2023-03-02T10:55:42.223780Z	warn	ca	ca request failed, starting attempt 3 in 434.158944ms
2023-03-02T10:55:42.658450Z	warn	ca	ca request failed, starting attempt 4 in 872.79127ms
2023-03-02T10:55:43.531589Z	warn	sds	failed to warm certificate: failed to generate workload certificate: create certificate: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp: lookup istiod.istio-system.svc on 10.96.0.10:53: read udp 10.22.0.5:47418->10.96.0.10:53: read: connection refused"
```

`kubectl logs default-domain-8r7ls -n knative-serving`:
```console
W0302 10:55:07.376158       1 client_config.go:617] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
{"level":"fatal","ts":1677754510.4817615,"logger":"fallback.default-domain","caller":"default-domain/main.go:199","msg":"Error finding gateway address","error":"Internal error occurred: failed calling webhook \"webhook.serving.knative.dev\": failed to call webhook: Post \"https://webhook.knative-serving.svc:443/defaulting?timeout=10s\": dial tcp 10.110.151.43:443: connect: no route to host","stacktrace":"main.main\n\tknative.dev/serving/cmd/default-domain/main.go:199\nruntime.main\n\truntime/proc.go:250"}
```

containerd and gVisor logs don't contain any error messages.

AFAIC, these are networking errors. Could someone please help me out? @anshalshukla @leokondrashov @ustiugov @shyamjesal

**Comment by ustiugov** [comment]
this is most likely not the first problem that occurs in the setup. 

Please provide more details on where you run it and describe the exact steps to reproduce. Then, you can ask @anshalshukla for help with logs; he has the most experience.

**Comment by CuriousGeorgiy** [comment]
> Please provide more details on where you run it and describe the exact steps to reproduce.

@ustiugov What details are needed?

I already described the exast steps to reproduce:
> I am following the standard clutser setup procedure described in quickstart guide, and I keep get the following errors with gVisor setup (both main branch and my branch).

This happens on all of the machines recommended for setting up vHive.

**Comment by leokondrashov** [comment]
I think that specific nodes that you run it, specific commands (quickstart guide is not a very specific thing, you can provide different arguments to scripts to choose runtime like firecracker, gVisor, or stargz deployment) are needed to reproduce.

Commands that I've provided in one of the previous comments are based on the quickstart guide but differ enough so one can easily miss something while trying to following the guide.

**Comment by anshalshukla** [comment]
As pointed out by Dmitrii I feel the issue is with the istio installation, specifically ingress-gateway but not sure about any solution

**Comment by CuriousGeorgiy** [comment]
@ustiugov I added the precise steps to reproduce the problems to my comment above.

**Comment by ustiugov** [comment]
Two things: 

why do you run firecracker-containerd? It's definitely redundant for the gvisor setup

Can you check the output of the setup scripts? Does istio installation succeeds? Any other errors?

Also, it would be better to reproduce this problem on a single node.

**Comment by CuriousGeorgiy** [comment]
@ustiugov 

> why do you run firecracker-containerd? It's definitely redundant for the gvisor setup
It's a typo, I didn't mean this line.

> Can you check the output of the setup scripts? Does istio installation succeeds? Any other errors?

As I mentioned, istio installation does not succeed.

> Also, it would be better to reproduce this problem on a single node.

As I mentioned, this problem is reproducible on a single node (both @anshalshukla and @leokondrashov managed to reproduce it).

File: configs/gvisor-containerd/config.toml [hunk 1]
```diff
@@ -5,6 +5,8 @@ state = "/run/gvisor-containerd"
   address = "/run/gvisor-containerd/containerd.sock"
 [plugins."io.containerd.runtime.v1.linux"]
   shim_debug = true
+[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc]
+  runtime_type = "io.containerd.runc.v1"
```
Comment (ustiugov) [inline configs/gvisor-containerd/config.toml:9]: as non-function components should run in the regular runc containers, I am not sure what these 2 lines are. I'd try removing them. 

Also, please check that only functions and nothing else runs in gvisor sandbox (top on the nodes would do to spot gvisor processes).

**Comment by ustiugov** [comment]
@CuriousGeorgiy if istio setup fails, this is our target for debugging. please proceed with the steps I mentioned above.

File: configs/gvisor-containerd/config.toml (hunk 1 above)
Comment (CuriousGeorgiy) [reply configs/gvisor-containerd/config.toml:9]: > as non-function components should run in the regular runc containers, I am not sure what these 2 lines are. I'd try removing them.

Removing these lines does not help (actually, I added these lines to try to make it work).

> Also, please check that only functions and nothing else runs in gvisor sandbox (top on the nodes would do to spot gvisor processes).

Not sure what you mean, but I filtered out the processes by `gvisor` and there's only `gvisor-containerd` running:
Screenshot 2023-03-08 at 15 27 24
Screenshot 2023-03-08 at 15 21 05

**Comment by CuriousGeorgiy** [comment]
@ustiugov

> if istio setup fails, this is our target for debugging.

IMHO, this should be debugged in scope of #557. Since single node setup works, we should focus on debugging gvisor in the scope of this PR.

> please proceed with the steps I mentioned above.

What particular steps?

**Comment by CuriousGeorgiy** [comment]
I have analyzed the log difference between the main branch and the PR branch during the following steps:

<details> <summary> main branch: </summary>
```console
git clone https://github.com/vhive-serverless/vhive.git

cd vhive

mkdir -p /tmp/vhive-logs

./scripts/cloudlab/setup_node.sh gvisor > >(tee -a /tmp/vhive-logs/setup_node.stdout) 2> >(tee -a /tmp/vhive-logs/setup_node.stderr >&2)

./scripts/cloudlab/start_onenode_vhive_cluster.sh gvisor > >(tee -a /tmp/vhive-logs/start_onenode_vhive_cluster.stdout) 2> >(tee -a /tmp/vhive-logs/start_onenode_vhive_cluster.stderr >&2)

source /etc/profile && pushd ./examples/deployer && go build && popd && ./examples/deployer/deployer
```
</details>

<details> <summary> PR branch: </summary>
```console
git clone https://github.com/CuriousGeorgiy/vhive.git

cd vhive

mkdir -p /tmp/vhive-logs

git checkout gh-683-upgrade-gvisor

./scripts/cloudlab/setup_node.sh gvisor > >(tee -a /tmp/vhive-logs/setup_node.stdout) 2> >(tee -a /tmp/vhive-logs/setup_node.stderr >&2)

./scripts/cloudlab/start_onenode_vhive_cluster.sh gvisor > >(tee -a /tmp/vhive-logs/start_onenode_vhive_cluster.stdout) 2> >(tee -a /tmp/vhive-logs/start_onenode_vhive_cluster.stderr >&2)

source /etc/profile && pushd ./examples/deployer && go build && popd && ./examples/deployer/deployer
```
</details>

The setup and startup script logs are identical.

The containerd and vHive orchestrator logs on the PR branch have additional log entries, the gVisor logs diverge at the tail.

<details> <summary> containerd PR branch logs: </summary>
```console
time="2023-03-10T02:14:27.031554014-07:00" level=info msg="StartContainer for \"d1f8136eac10d13a0c78476980cccaeede748072885b02828685983afa61cae4\""
time="2023-03-10T02:14:27.148468619-07:00" level=info msg="StartContainer for \"d1f8136eac10d13a0c78476980cccaeede748072885b02828685983afa61cae4\" returns successfully"
time="2023-03-10T02:15:28.074415426-07:00" level=info msg="StopContainer for \"d1f8136eac10d13a0c78476980cccaeede748072885b02828685983afa61cae4\" with timeout 300 (s)"
time="2023-03-10T02:15:28.076026423-07:00" level=info msg="Stop container \"d1f8136eac10d13a0c78476980cccaeede748072885b02828685983afa61cae4\" with signal terminated"
time="2023-03-10T02:15:58.081082392-07:00" level=info msg="StopContainer for \"3e766a54c6365c33c6317425a6321dc5f7d13f45c54b8dd9c9dfbf41b1a93262\" with timeout 300 (s)"
time="2023-03-10T02:15:58.081724337-07:00" level=info msg="Stop container \"3e766a54c6365c33c6317425a6321dc5f7d13f45c54b8dd9c9dfbf41b1a93262\" with signal terminated"
time="2023-03-10T02:15:58.115683927-07:00" level=info msg="shim disconnected" id=d1f8136eac10d13a0c78476980cccaeede748072885b02828685983afa61cae4
time="2023-03-10T02:15:58.115766496-07:00" level=warning msg="cleaning up after shim disconnected" id=d1f8136eac10d13a0c78476980cccaeede748072885b02828685983afa61cae4 namespace=k8s.io
time="2023-03-10T02:15:58.115789289-07:00" level=info msg="cleaning up dead shim"
time="2023-03-10T02:15:58.129310162-07:00" level=warning msg="cleanup warnings time=\"2023-03-10T02:15:58-07:00\" level=info msg=\"starting signal loop\" namespace=k8s.io pid=32459 runtime=io.containerd.runc.v2\n"
time="2023-03-10T02:15:58.130505751-07:00" level=info msg="StopContainer for \"d1f8136eac10d13a0c78476980cccaeede748072885b02828685983afa61cae4\" returns successfully"
time="2023-03-10T02:15:58.185056761-07:00" level=info msg="shim disconnected" id=3e766a54c6365c33c6317425a6321dc5f7d13f45c54b8dd9c9dfbf41b1a93262
time="2023-03-10T02:15:58.185153573-07:00" level=warning msg="cleaning up after shim disconnected" id=3e766a54c6365c33c6317425a6321dc5f7d13f45c54b8dd9c9dfbf41b1a93262 namespace=k8s.io
time="2023-03-10T02:15:58.185178769-07:00" level=info msg="cleaning up dead shim"
time="2023-03-10T02:15:58.200865016-07:00" level=warning msg="cleanup warnings time=\"2023-03-10T02:15:58-07:00\" level=info msg=\"starting signal loop\" namespace=k8s.io pid=32489 runtime=io.containerd.runc.v2\n"
time="2023-03-10T02:15:58.201871987-07:00" level=info msg="StopContainer for \"3e766a54c6365c33c6317425a6321dc5f7d13f45c54b8dd9c9dfbf41b1a93262\" returns successfully"
time="2023-03-10T02:15:58.202904198-07:00" level=info msg="StopPodSandbox for \"01c5c2568835c2a877993150ebdec21dd8920f36d5a694c12c7dffc2ceeaa23e\""
time="2023-03-10T02:15:58.203032786-07:00" level=info msg="Container to stop \"d1f8136eac10d13a0c78476980cccaeede748072885b02828685983afa61cae4\" must be in running or unknown state, current state \"CONTAINER_EXITED\""
time="2023-03-10T02:15:58.203068032-07:00" level=info msg="Container to stop \"3e766a54c6365c33c6317425a6321dc5f7d13f45c54b8dd9c9dfbf41b1a93262\" must be in running or unknown state, current state \"CONTAINER_EXITED\""
time="2023-03-10T02:15:58.297232468-07:00" level=info msg="shim disconnected" id=01c5c2568835c2a877993150ebdec21dd8920f36d5a694c12c7dffc2ceeaa23e
time="2023-03-10T02:15:58.297287578-07:00" level=warning msg="cleaning up after shim disconnected" id=01c5c2568835c2a877993150ebdec21dd8920f36d5a694c12c7dffc2ceeaa23e namespace=k8s.io
time="2023-03-10T02:15:58.297301377-07:00" level=info msg="cleaning up dead shim"
time="2023-03-10T02:15:58.308435103-07:00" level=warning msg="cleanup warnings time=\"2023-03-10T02:15:58-07:00\" level=info msg=\"starting signal loop\" namespace=k8s.io pid=32528 runtime=io.containerd.runc.v2\n"
time="2023-03-10T02:15:58.517985042-07:00" level=info msg="TearDown network for sandbox \"01c5c2568835c2a877993150ebdec21dd8920f36d5a694c12c7dffc2ceeaa23e\" successfully"
time="2023-03-10T02:15:58.518050557-07:00" level=info msg="StopPodSandbox for \"01c5c2568835c2a877993150ebdec21dd8920f36d5a694c12c7dffc2ceeaa23e\" returns successfully"
time="2023-03-10T02:15:58.673958466-07:00" level=info msg="RemoveContainer for \"d1f8136eac10d13a0c78476980cccaeede748072885b02828685983afa61cae4\""
time="2023-03-10T02:15:58.679129051-07:00" level=info msg="RemoveContainer for \"d1f8136eac10d13a0c78476980cccaeede748072885b02828685983afa61cae4\" returns successfully"
time="2023-03-10T02:15:58.681979391-07:00" level=info msg="RemoveContainer for \"3e766a54c6365c33c6317425a6321dc5f7d13f45c54b8dd9c9dfbf41b1a93262\""
time="2023-03-10T02:15:58.687500857-07:00" level=info msg="RemoveContainer for \"3e766a54c6365c33c6317425a6321dc5f7d13f45c54b8dd9c9dfbf41b1a93262\" returns successfully"
time="2023-03-10T02:15:58.688453451-07:00" level=error msg="ContainerStatus for \"d1f8136eac10d13a0c78476980cccaeede748072885b02828685983afa61cae4\" failed" error="rpc error: code = NotFound desc = an error occurred when try to find container \"d1f8136eac10d13a0c78476980cccaeede748072885b02828685983afa61cae4\": not found"
time="2023-03-10T02:15:58.689549044-07:00" level=error msg="ContainerStatus for \"3e766a54c6365c33c6317425a6321dc5f7d13f45c54b8dd9c9dfbf41b1a93262\" failed" error="rpc error: code = NotFound desc = an error occurred when try to find container \"3e766a54c6365c33c6317425a6321dc5f7d13f45c54b8dd9c9dfbf41b1a93262\": not found"
time="2023-03-10T02:15:58.690369091-07:00" level=error msg="ContainerStatus for \"d1f8136eac10d13a0c78476980cccaeede748072885b02828685983afa61cae4\" failed" error="rpc error: code = NotFound desc = an error occurred when try to find container \"d1f8136eac10d13a0c78476980cccaeede748072885b02828685983afa61cae4\": not found"
time="2023-03-10T02:15:58.691214088-07:00" level=error msg="ContainerStatus for \"3e766a54c6365c33c6317425a6321dc5f7d13f45c54b8dd9c9dfbf41b1a93262\" failed" error="rpc error: code = NotFound desc = an error occurred when try to find container \"3e766a54c6365c33c6317425a6321dc5f7d13f45c54b8dd9c9dfbf41b1a93262\": not found"
time="2023-03-10T02:16:38.381544370-07:00" level=info msg="StopPodSandbox for \"01c5c2568835c2a877993150ebdec21dd8920f36d5a694c12c7dffc2ceeaa23e\""
time="2023-03-10T02:16:38.394076537-07:00" level=info msg="TearDown network for sandbox \"01c5c2568835c2a877993150ebdec21dd8920f36d5a694c12c7dffc2ceeaa23e\" successfully"
time="2023-03-10T02:16:38.394129265-07:00" level=info msg="StopPodSandbox for \"01c5c2568835c2a877993150ebdec21dd8920f36d5a694c12c7dffc2ceeaa23e\" returns successfully"
time="2023-03-10T02:16:38.395241445-07:00" level=info msg="RemovePodSandbox for \"01c5c2568835c2a877993150ebdec21dd8920f36d5a694c12c7dffc2ceeaa23e\""
time="2023-03-10T02:16:38.395330196-07:00" level=info msg="Forcibly stopping sandbox \"01c5c2568835c2a877993150ebdec21dd8920f36d5a694c12c7dffc2ceeaa23e\""
time="2023-03-10T02:16:38.442629007-07:00" level=info msg="TearDown network for sandbox \"01c5c2568835c2a877993150ebdec21dd8920f36d5a694c12c7dffc2ceeaa23e\" successfully"
time="2023-03-10T02:16:38.447710333-07:00" level=info msg="RemovePodSandbox \"01c5c2568835c2a877993150ebdec21dd8920f36d5a694c12c7dffc2ceeaa23e\" returns successfully"
```
</details>

<details> <summary> gVisor main branch logs: </summary>
```console
time="2023-03-10T02:03:37.628278567-07:00" level=debug msg="registering ttrpc server"
time="2023-03-10T02:03:37.628379841-07:00" level=debug msg="serving api on socket" socket="[inherited from parent]"
time="2023-03-10T02:03:37.628409199-07:00" level=info msg="starting signal loop" namespace=default path=/run/gvisor-containerd/io.containerd.runtime.v2.task/default/1 pid=31710
time="2023-03-10T02:03:37.631737846-07:00" level=debug msg="Executing: [runsc --root=/run/containerd/runsc/default --log=/run/gvisor-containerd/io.containerd.runtime.v2.task/default/1/log.json --log-format=json create --bundle /run/gvisor-containerd/io.containerd.runtime.v2.task/default/1 --pid-file /run/gvisor-containerd/io.containerd.runtime.v2.task/default/1/init.pid 1]"
```
</details>

<details> <summary> gVisor PR branch logs: </summary>
```console
time="2023-03-10T02:14:24.306573191-07:00" level=info msg="starting signal loop" namespace=default path=/run/gvisor-containerd/io.containerd.runtime.v2.task/default/1 pid=31401
time="2023-03-10T02:14:24.309960086-07:00" level=error msg="Can't get pod uid" error="no sandbox log path annotation"
```
</details>

<details> <summary> vHive orchestrator PR branch logs: </summary>
```console
time="2023-03-10T02:15:58-07:00" level=error msg="failed to stop container" error="failed to find a active container with id d1f8136eac10d13a0c78476980cccaeede748072885b02828685983afa61cae4"
time="2023-03-10T02:15:58-07:00" level=error msg="failed to teardown network: plugin type=\"bridge\" name=\"mynet\" failed (delete): netplugin failed with no error message: signal: killed"
time="2023-03-10T02:15:58-07:00" level=error msg="failed to stop container" error="failed to kill task: context canceled"
```
</details>

File: scripts/install_stock.sh [hunk 2]
```diff
@@ -37,7 +37,7 @@ wget --continue --quiet https://github.com/opencontainers/runc/releases/download
 mv runc.amd64 runc
 sudo install -D -m0755 runc /usr/local/sbin/runc
 
-wget --continue --quiet https://storage.googleapis.com/gvisor/releases/release/20210622/x86_64/runsc 
+wget --continue --quiet https://storage.googleapis.com/gvisor/releases/release/20221026/x86_64/runsc
```
Comment (ustiugov) [inline scripts/install_stock.sh:40]: @CuriousGeorgiy why didn't you take a newer version of gVisor from 2023?

Comment (CuriousGeorgiy) [reply scripts/install_stock.sh:40]: Originally, I took the latest version — it didn't work. So I tried using older versions, but it didn't help.

**Comment by leokondrashov** [comment]
I've revisited comments here and found out that the issue that was blocking the PR is closely resembling the issue solved by #785. I think that this as well might be the issue here, since the problem was not caused by specific runtime, but rather network bridges vHive has created.

**Comment by CuriousGeorgiy** [comment]
@leokondrashov rebased the branch, let's see if the gVisor CRI workflow passes.

**Comment by CuriousGeorgiy** [comment]
@ustiugov this PR does not affect firecracker, so I guess we can ignore the firecracker CRI workflow failure

**Comment by leokondrashov** [comment]
Does our CI test for multi node setup? I don't think so, so it should be checked manually in addition to tests.

**Comment by ustiugov** [comment]
I agree with Georgiy, can merge.

**Comment by ustiugov** [comment]
if there going to be issues with the multi-node setup, please raise an issue. Istio failure should not be related to gvisor...

**Pull request merged**